### PR TITLE
fix: add delay registration to default scope handler

### LINKED ISSUE #49: Default command handler overshadowing every other scope
Creating and registering a command with a default handler will always overshadow any other scope. In the following snippet, even when conditions are met (e.g: being admin in a group chat), the response will always be "hello from default scope".
```ts
myCommands
  .command(
    "default",
    "Default command",
    (ctx) => ctx.reply("Hello from default scope"),
  )
  .addToScope(
    { type: "all_chat_administrators" },
    (ctx) => ctx.reply("Hello, admin!"),
  )
  .addToScope(
    { type: "all_group_chats" },
    (ctx) => ctx.reply("Hello, group chat!"),
  );
```
The order of `.addToScope` registration should affect indeed the handling priority, while the default handler should always be last.

### PR DESCRIPTION
**Pull request by roziscoding**
Fixes #49

### CODE REVIEW

**Review by carafelix — APPROVED** [review]
Simple and clean. LGTM

**Comment by KnorpelSenf** [comment]
https://github.com/grammyjs/commands/pull/50#discussion_r1831294674 needs to be fixed before tests are useful

File: src/command.ts [hunk 1]
```diff
@@ -445,6 +446,9 @@ export class Command<C extends Context = Context> implements MiddlewareObj<C> {
   }
 
   middleware() {
-    return this._composer.middleware();
+    return new Composer<C>()
```
[the remark this answers is not in the document]
Comment (KnorpelSenf) [reply src/command.ts]: LGTM now

**Review by carafelix — APPROVED** [review]
The man as spoken

**Pull request merged**